pkg/controller/operators/catalog/operator.go

```diff
@@ -276,19 +273,39 @@ func (o *Operator) handleDeletion(obj interface{}) {
 	}
 
 	if owner := ownerutil.GetOwnerByKind(ownee, v1alpha1.CatalogSourceKind); owner != nil {
-		o.catSrcQueue.AddRateLimited(fmt.Sprintf("%s/%s", ownee.GetNamespace(), owner.Name))
+		if err := o.catSrcQueueSet.Requeue(owner.Name, ownee.GetNamespace()); err != nil {
+			o.Log.Warn(err.Error())
+		}
 	}
 }
 
 func (o *Operator) handleCatSrcDeletion(obj interface{}) {
-	if catsrc, ok := obj.(*v1alpha1.CatalogSource); ok {
-		sourceKey := resolver.CatalogKey{Name: catsrc.GetName(), Namespace: catsrc.GetNamespace()}
-		func() {
-			o.sourcesLock.Lock()
-			defer o.sourcesLock.Unlock()
-			delete(o.sources, sourceKey)
-		}()
-		o.Log.WithField("source", sourceKey).Info("removed client for deleted catalogsource")
+	catsrc, ok := obj.(metav1.Object)
+	if !ok {
+		if !ok {
+			tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+				return
+			}
+
+			catsrc, ok = tombstone.Obj.(metav1.Object)
+			if !ok {
+				utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a Namespace %#v", obj))
+				return
+			}
+		}
+	}
+	sourceKey := resolver.CatalogKey{Name: catsrc.GetName(), Namespace: catsrc.GetNamespace()}
+	func() {
+		o.sourcesLock.Lock()
+		defer o.sourcesLock.Unlock()
+		delete(o.sources, sourceKey)
+	}()
+	o.Log.WithField("source", sourceKey).Info("removed client for deleted catalogsource")
+
+	if err := o.catSrcQueueSet.Remove(sourceKey.Name, sourceKey.Namespace); err != nil {
```

Comment:
I don't think we want to try to remove the CatalogSource queue for the namespace every time we observe a CatalogSource deletion.